packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -155,7 +156,81 @@ export async function forceLightningLwcPreview(sourceUri: vscode.Uri) {
     return;
   }
 
-  await selectPlatformAndExecute(startTime, componentName);
+  await executePreview(startTime, componentName, resourcePath);
+}
+
+/**
+ * Performs the action of previewing the LWC. It takes care of prompting the user
+ * and gathering all info needed to preview the LWC. This includes prompting the user
+ * to select a platform, a target device, a target native app (or borwser), etc.
+ * Previewing on Android or iOS are handled by the @salesforce/lwc-dev-mobile sfdx package.
+ *
+ * @param startTime start time of the preview command
+ * @param componentName name of the lwc
+ * @param resourcePath path to the lwc
+ */
+async function executePreview(
+  startTime: [number, number],
+  componentName: string,
+  resourcePath: string
+) {
+  // 1. Prompt user to select a platform
+  const platformSelection = await selectPlatform();
+  if (!platformSelection) {
+    vscode.window.showWarningMessage(
+      nls.localize('force_lightning_lwc_operation_cancelled')
+    );
+    return;
+  }
+
+  if (platformSelection.id === PreviewPlatformType.Desktop) {
+    await startServer(true, componentName, startTime);
+    return;
+  }
+
+  const commandCancelledMessage = nls.localize(
```

Comment:
This may as well move somewhere above [line 179](https://github.com/forcedotcom/salesforcedx-vscode/pull/2585/files#diff-785bdec339e64f40207b4e9dcae14dd4R179), since it's dereferenced there too.